File: data/db_models/IngredientRowItem.ts
import { DataTypes, Model, Optional, Sequelize } from "sequelize";
import { IngredientRow } from "./IngredientRowData";


interface IngredientRowItemAttributes {
    id: number;
    rowId: number;
    itemId: number;
    name: string;
    amt: number;
    unit: string;
    water_footprint: number;
    unselectedBgImageUrl: string;
    selectedBgImageUrl: string;
    sampleImageUrl: string;
    sampleImageSize: number;
    scaleFactor: number;
    iconScalefactor: number;
    cornerType: string;
    doneXOffSet: number;
    doneYOffSet: number;
    pluseXOffSet: number;
    pluseYOffSet: number;
    minusXOffSet: number;
    minusYOffSet: number;
    xOffset: number;
    yOffset: number;
}

interface IngredientRowItemCreationAttributes extends Optional<IngredientRowItemAttributes, 'id'> {}

export class IngredientRowItem extends Model<IngredientRowItemAttributes, IngredientRowItemCreationAttributes> {
    static belongsTo(IngredientRow: typeof IngredientRow, arg1: { foreignKey: string; }) {
      throw new Error('Method not implemented.');
    }
    declare id: number;
    declare itemId: number;
    declare rowId: number;
    declare name: string;
    declare amt: number;
    declare unit: string;
    declare water_footprint: number;
    declare unselectedBgImageUrl: string;
    declare selectedBgImageUrl: string;
    declare sampleImageUrl: string;
    declare sampleImageSize: number;
    declare scaleFactor: number;
    declare iconScalefactor: number;
    declare cornerType: string;
    declare doneXOffSet: number;
    declare doneYOffSet: number;
    declare pluseXOffSet: number;
    declare pluseYOffSet: number;
    declare minusXOffSet: number;
    declare minusYOffSet: number;
    declare xOffset: number;
    declare yOffset: number;
  
    public readonly createdAt!: Date;
    public readonly updatedAt!: Date;
  
  }
  
  export function initIngredientRowItem(sequelize: Sequelize) {
    
    IngredientRowItem.init(
        {
          id: {
            type: DataTypes.INTEGER,
            autoIncrement: true,
            unique: true,
            primaryKey: true,
          },
          itemId: {
            type: DataTypes.INTEGER,
            allowNull: false,
          
          },
          rowId: {
            type: DataTypes.INTEGER,
            allowNull: false,
            references: {
                model: IngredientRow,
                key: 'id',
              },
          },
          name: {
            type: DataTypes.STRING(255),
            allowNull: false,
            unique: true
          },
          amt: {
            type: DataTypes.DECIMAL,
            allowNull: false
          },
          unit: {
            type: DataTypes.STRING,
            allowNull: false
          },
          water_footprint: {
            type: DataTypes.DECIMAL,
            allowNull: false
          },
          unselectedBgImageUrl: {
            type: DataTypes.STRING,
            allowNull: false
          }, 
          selectedBgImageUrl: {
            type: DataTypes.STRING,
            allowNull: false
          }, 
          sampleImageUrl: {
            type: DataTypes.STRING,
            allowNull: false
          },
          sampleImageSize: {
            type: DataTypes.DECIMAL,
            allowNull: false
          },
          scaleFactor: {
            type: DataTypes.DECIMAL,
            allowNull: false
          },
          iconScalefactor: {
            type: DataTypes.DECIMAL,
            allowNull: false
          },
          cornerType: {
            type: DataTypes.STRING,
            allowNull: false
          },
          doneXOffSet: {
            type: DataTypes.DECIMAL,
            allowNull: false
          },
          doneYOffSet: {
            type: DataTypes.DECIMAL,
            allowNull: false
          },
          pluseXOffSet: {
            type: DataTypes.DECIMAL,
            allowNull: false
          },
          pluseYOffSet: {
            type: DataTypes.DECIMAL,
            allowNull: false
          },
          minusXOffSet: {
            type: DataTypes.DECIMAL,
            allowNull: false
          },
          minusYOffSet: {
            type: DataTypes.DECIMAL,
            allowNull: false
          },
          xOffset: {
            type: DataTypes.DECIMAL,
            allowNull: false,
          },
          yOffset: {
            type: DataTypes.DECIMAL,
            allowNull: false,
          
          }
        },
        {
          sequelize: sequelize,
          tableName: 'ingredient_row_item',
          timestamps: true
        }
      ).sync({ alter: true })

      
}
  

File: app/src/main/java/org/arghyam/puddle/data/dto/responses/water_ft_calculator/IngredientRowItem.kt
package org.arghyam.puddle.data.dto.responses.water_ft_calculator


import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class IngredientRowItem(
    @SerialName("amt")
    val amt: Int,
    @SerialName("cornerType")
    val cornerType: String,
    @SerialName("createdAt")
    val createdAt: String,
    @SerialName("doneXOffSet")
    val doneXOffSet: Float,
    @SerialName("doneYOffSet")
    val doneYOffSet: Float,
    @SerialName("iconScalefactor")
    val iconScaleFactor: Float,
    @SerialName("id")
    val id: Int,
    @SerialName("itemId")
    val itemId: Int,
    @SerialName("minusXOffSet")
    val minusXOffSet: Float,
    @SerialName("minusYOffSet")
    val minusYOffSet: Float,
    @SerialName("name")
    val name: String,
    @SerialName("pluseXOffSet")
    val pluseXOffSet: Float,
    @SerialName("pluseYOffSet")
    val pluseYOffSet: Float,
    @SerialName("rowId")
    val rowId: Int,
    @SerialName("sampleImageSize")
    val sampleImageSize: Float,
    @SerialName("sampleImageUrl")
    val sampleImageUrl: String,
    @SerialName("scaleFactor")
    val scaleFactor: Float,
    @SerialName("selectedBgImageUrl")
    val selectedBgImageUrl: String,
    @SerialName("unit")
    val unit: String,
    @SerialName("unselectedBgImageUrl")
    val unselectedBgImageUrl: String,
    @SerialName("updatedAt")
    val updatedAt: String,
    @SerialName("xOffset")
    val xOffset: Float,
    @SerialName("yOffset")
    val yOffset: Float
)